DigitalLearningSolutions.Data.Tests/DataServices/SelfAssessmentDataServiceTests/CompetencyDataServiceTests.cs

```diff
@@ -0,0 +1,387 @@
+                name: "Taking an active role in my own learning is the most important thing that affects my digital literacy skills development",
+                description: "Taking an active role in my own learning is the most important thing that affects my digital literacy skills development",
+                assessmentQuestions: new List<AssessmentQuestion>()
+                {
+                    SelfAssessmentHelper.CreateDefaultAssessmentQuestion(
+                        id: 3,
+                        maxValueDescription: "Strongly agree",
+                        minValueDescription: "Strongly disagree",
+                        question: "To what extent do you agree",
+                       assessmentQuestionInputTypeID: 1
+                    )
+                });
+
+            // When
+            var result = selfAssessmentDataService.GetNthCompetency(32, SelfAssessmentId, CandidateId);
+
+            // Then
+            result.Should().BeEquivalentTo(expectedCompetency);
+        }
+
+        [Test]
+        public void GetNthCompetency_returns_null_when_reached_end_of_assessment()
+        {
+            // When
+            var result = selfAssessmentDataService.GetNthCompetency(33, SelfAssessmentId, CandidateId);
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void GetNthCompetency_returns_null_when_n_zero()
+        {
+            // When
+            var result = selfAssessmentDataService.GetNthCompetency(0, SelfAssessmentId, CandidateId);
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void GetNthCompetency_returns_null_when_n_negative()
+        {
+            // When
+            var result = selfAssessmentDataService.GetNthCompetency(-1, SelfAssessmentId, CandidateId);
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void GetNthCompetency_returns_null_when_invalid_candidate()
+        {
+            // When
+            var result = selfAssessmentDataService.GetNthCompetency(1, SelfAssessmentId, 1);
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void GetNthCompetency_returns_null_when_invalid_assessment()
+        {
+            // When
+            var result = selfAssessmentDataService.GetNthCompetency(1, 2, 1);
+
+            // Then
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void GetNthCompetency_gets_the_latest_result()
+        {
+            // Given
+            const int competencyId = 2;
+            const int assessmentQuestionId = 2;
+            const int result = 5;
+
+            using (new TransactionScope())
+            {
+                // When
+                selfAssessmentDataService.SetResultForCompetency(competencyId, SelfAssessmentId, CandidateId, assessmentQuestionId, result + 1, null);

```

Comment:
Across this file there are a whole bunch of lines that are rather long due a lot of parameters. Running the formatter here would sort this out, but would make the tests significantly longer line-wise, and thus harder to read in my mind. Thoughts?

Reply:
I think it's best to just run it (and have done so).